darts/timeseries.py

```diff
@@ -140,156 +582,251 @@ def pd_dataframe(self, copy=True) -> pd.DataFrame:
         Returns
         -------
         pandas.DataFrame
-            The Pandas Dataframe underlying this time series
+            The Pandas DataFrame representation of this time series
         """
+        self._assert_deterministic()
         if copy:
-            return self._df.copy()
+            return pd.DataFrame(self._xa[:, :, 0].values.copy(),
+                                index=self._time_index.copy(),
+                                columns=self._xa.get_index('component').copy())
         else:
-            return self._df
+            return pd.DataFrame(self._xa[:, :, 0].values,
+                                index=self._time_index,
+                                columns=self._xa.get_index('component'))
 
-    def columns(self):
-        return self.pd_dataframe().columns
+    def quantile_df(self, quantile=0.5) -> pd.DataFrame:
+        """
+        Returns a Pandas DataFrame containing the single desired quantile of each component (over the samples).
+        Each of the series components will appear as a column in the DataFrame. The column will be named
+        "<component>_X", where "<component>" is the column name corresponding to this component, and "X"
+        is the quantile value.
+        The quantile columns represent the marginal distributions of the components of this series.
+
+        This works only on stochastic series (i.e., with more than 1 sample)
+
+        Parameters
+        ----------
+        quantile
+            The desired quantile value. The value must be represented as a fraction
+            (between 0 and 1 inclusive). For instance, `0.5` will return a DataFrame
+            containing the median of the (marginal) distribution of each component.
+
+        Returns
+        -------
+        pandas.DataFrame
+            The Pandas DataFrame containing the desired quantile for each component.
+        """
+        self._assert_stochastic()
+        raise_if_not(0 <= quantile <= 1,
+                     'The quantile values must be expressed as fraction (between 0 and 1 inclusive).', logger)
+
+        # column names
+        cnames = list(map(lambda s: s + '_{}'.format(quantile), self.columns))
```

Comment:
According to here: https://stackoverflow.com/questions/1247486/list-comprehension-vs-map map is slower when using lambda functions compared to list comprehension. Also I'd consider list comprehension more pythonic and readable

Reply:
You're right... not sure why I did this :D